### DISCUSSION TITLE
Include parameters in header and footer in Quarto PDF?

### DISCUSSION THREAD

**Discussion by jhelvy** [quarto-dev/quarto-cli#5388] [Q&A] [answered]
I asked this question on [SO](https://stackoverflow.com/questions/76158024/include-parameters-in-header-and-footer-in-quarto-rmarkdown-pdf) as well. Is it possible to include parameters in the header and / or footer text of a Quarto PDF document?

[This issue](https://stackoverflow.com/questions/75132359/header-and-footer-in-quarto-qmd-to-pdf) shows how to include general text in the header and footer, but I want to include a *parameter* in the header or footer that I can pass while rendering.

In the example below, the parameter `name` can be rendered in the main body, but I do not know if it is possible to include it in the header. 

```
---
title: "Untitled"
format:
  pdf:
    include-in-header:
      text: |
        \usepackage{scrlayer-scrpage}
        \rohead{Header text}
        \lofoot{Footer text}
params:
  name: "John"
---

# Chapter ONE

My name is `r params$name`

\newpage

# Chapter TWO
```

**Comment by cscheid** [comment]
(EDITED: I just realized our `meta` shortcode works on YAML itself.)

Use a shortcode: `{{< meta params.name >}}`

**Reply by jhelvy** [reply]
That's what I was afraid of :(
Hopefully someone can find a hack to implement this.

**Reply by cscheid** [reply]
Actually it just occurred to me that the `meta` shortcode _should_ work. You can't run arbitrary code but you can grab information from the yaml metadata. I've tested this on HTML, but that should work in general:

````
---
title: "Untitled"
format:
  html:
    include-in-header:
      text: |
        <meta name="key" content="{{< meta params.name >}}"/>
params:
  name: foo
---

{{< meta params.name >}}
````

I then see this in the HTML header:

<img width="1625" alt="image" src="https://user-images.githubusercontent.com/285675/235778944-dc6ff62f-46a6-4b97-a0f9-508b9e3abf8a.png">

**Comment by baptiste** [comment, marked as answer]
For latex, you can use this workaround: have the header/footer refer to a macro, that you define later on in the document

````
---
title: "Untitled"
format:
  pdf:
    keep-tex: true
    include-in-header:
      text: |
        \usepackage{scrlayer-scrpage}
        \rohead{\myname}
params:
  name: "John"

engine: knitr
---

# Chapter ONE

My name is `r params$name`

\newcommand{\myname}{`r params$name`}


\newpage

# Chapter TWO


````

**Reply by cscheid** [reply]
That's clever!

**Reply by jhelvy** [reply]
I've been trying to come up with something like this - beautiful!

**Reply by cscheid** [reply]
@jhelvy There may be a native solution for the simple case of resolving a parameter, but @baptiste's trick is going to be more general, because it lets you refer to anything that's evaluated.
